modules/xml/test/src/smithy4s/xml/XmlCodecSpec.scala

```diff
@@ -481,43 +494,73 @@ object XmlDecodingSpec extends SimpleIOSuite {
     val xml = """|<Bar>
                  |   <x>1</x>
                  |</Bar>""".stripMargin
-    decodeDocument[Foo](xml).attempt.map { result =>
-      expect.same(
-        result,
-        Left(
-          XmlDecodeError(XPath.root, "Expected Foo XML root element, got Bar")
+    parseDocument(xml)
+      .flatMap(decodeDocument[Foo](_))
+      .attempt
+      .map { result =>
+        expect.same(
+          result,
+          Left(
+            XmlDecodeError(XPath.root, "Expected Foo XML root element, got Bar")
+          )
         )
-      )
-    }
+      }
   }
 
-  def checkContent[A: Schema](xml: String, expected: A)(implicit
+  def checkContent[A: Schema](xmlString: String, expected: A)(implicit
       loc: SourceLocation
   ): IO[Expectations] = {
-    decodeContent[A](xml).map(result => expect.same(result, expected))
+    parseDocument(xmlString).flatMap { document =>
+      val decodingChecks = decodeContent[A](document)
+        .map(result => expect.same(result, expected))
+
+      import cats.Show
+      implicit val showXmlDocument: Show[XmlDocument] = new Show[XmlDocument] {
+        def show(xmlDocument: XmlDocument): String =
+          XmlDocument.documentEventifier
+            .eventify(xmlDocument)
+            .compile
+            .toVector
+            .map(_.show)
+            .mkString("")
```

Comment:
```suggestion
            .through(render())
            .compile
            .string
```

Has the advantage of using (you can opt-out) self-closing tags for more concise output. Added recently in fs2-data 1.5.1.